tinygrad/codegen/uops.py

```diff
@@ -294,14 +294,10 @@ def rewrite(u:UOp) -> UOp:
         changed += recurse_cnt
         # NOTE: this changes UOp, so we have to delete caches
         up.vin = tuple(rewrite(x) for x in up.vin)
-        try: del up.parents
-        except AttributeError: pass
-        try: del up.cmp_tuple
-        except AttributeError: pass
+        if 'parents' in up.__dict__: delattr(up, 'parents')
+        if 'cmp_tuple' in up.__dict__: delattr(up, 'cmp_tuple')
         # replace with cached nodes
-        if found:=self.nodes.get(key:=up.tuple()): return found
-        self.nodes[key] = up
-        return up
+        return self.nodes.setdefault(up.tuple(), up)
```

Comment:
Interesting. Do we have this pattern elsewhere in the code?

Reply:
Yes but might not be usefull everywhere (for perf) as the second argument is evaluated :
```python
>>> a = {"a": 1}
>>> a.setdefault("a", print("b"))
b
1
```
So it some case it's better to keep additional `if` to avoid extra compute